### PR TITLE
fix: correctly handle prefixed enums and static instance fields for prefer-moving-to-variable

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

- [ ] Documentation update
- [X] Bug fix
- [ ] New rule
- [ ] Changes an existing rule
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

### CODE REVIEW

**Review by dkrutskikh — APPROVED** [review]

File: test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'example.dart' as prefix;
+import 'generics_example.dart';
+
+void main() {
+  AnotherEnum.anotherValue;
+  AnotherEnum.firstValue;
+
+  prefix.SomeValue.firstValue;
```
Comment (jacob314) [inline test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: try referencing prefix.SomeValue.firstValue
repeatedly
I believe that would match the case where I was seeing false positives.

Comment (incendial) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: Yes, that's it! Thank you!

Comment (incendial) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: Hm, actually, if I duplicate the line 5 `AnotherEnum.anotherValue;` the rule will also trigger. So it's not connected to prefixes, maybe you remember another cases?

Comment (jacob314) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: The strange thing is removing the prefixes did seem to fix the issue in DevTools.  That said, I wouldn't want the rule to trigger for duplicated `AnotherEnum.anotherValue` case either.  Specifying the enum value name twice seems more idiomatic than defining an unneeded local variable.

Comment (incendial) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: Agree. And maybe the same should work for the class level static variables. You probably wouldn't want to assign this type of calls to a variable.

Comment (jacob314) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart:8]: Agree that I wouldn't want this firing for class to class level static variables either.

Comment (incendial) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_moving_to_variable/examples/prefix_example.dart]: Sorry for misleading, it was a problem exactly with the prefixes 🤦 
Now fixed, thank you!

**Review by jacob314 — APPROVED** [review]
lgtm

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**